Basic app install and app update commands

nblumhardt opened the pull request:
```
seqcli app install [<args>]

Install an app package

Arguments:
      --package-id=VALUE     The package id of the app to install
      --version=VALUE        The package version to install; the default is to
                               install the latest version
      --feed-id=VALUE        The id of the NuGet feed to install the package
                               from; may be omitted if only one feed is
                               configured
  -s, --server=VALUE         The URL of the Seq server; by default the `
                               connection.serverUrl` config value will be used
  -a, --apikey=VALUE         The API key to use when connecting to the server;
                               by default the `connection.apiKey` config
                               value will be used
      --profile=VALUE        A connection profile to use; by default the `
                               connection.serverUrl` and `connection.apiKey`
                               config values will be used
      --json                 Print output in newline-delimited JSON (the
                               default is plain text)
      --no-color             Don't colorize text output
      --force-color          Force redirected output to have ANSI color (
                               unless `--no-color` is also specified)
      --verbose              Print verbose output to `STDERR`
```

```
seqcli app update [<args>]

Update an installed app package

Arguments:
  -i, --id=VALUE             The id of a single installed app to update
  -n, --name=VALUE           The name of the installed app to update
      --all                  Update all installed apps; not compatible with `-
                               i` or `-n`
      --version=VALUE        The package version to update to; the default is
                               to update to the latest version in the
                               associated feed
      --force                Update the app even if the target version is
                               already installed
  -s, --server=VALUE         The URL of the Seq server; by default the `
                               connection.serverUrl` config value will be used
  -a, --apikey=VALUE         The API key to use when connecting to the server;
                               by default the `connection.apiKey` config
                               value will be used
      --profile=VALUE        A connection profile to use; by default the `
                               connection.serverUrl` and `connection.apiKey`
                               config values will be used
      --json                 Print output in newline-delimited JSON (the
                               default is plain text)
      --no-color             Don't colorize text output
      --force-color          Force redirected output to have ANSI color (
                               unless `--no-color` is also specified)
      --verbose              Print verbose output to `STDERR`
```

liammclennan reviewed (approved):
Some funny interactions for the update command. The user is supposed to specify exactly one of --id, --name, --all but this isn't clear.

What happens when the user specifies --all and --version is not likely to be what the user expects so I think this should be an error.

nblumhardt commented:
Thanks for the great review @liammclennan. I've updated the `seqcli help <command>` output to include an example - hopefully this improves the first issue you noted (providing more examples would definitely be nice here):

```
seqcli app install [<args>]

Install an app package

Example:
  seqcli app install --package-id 'Seq.App.JsonArchive'

Arguments:
      --package-id=VALUE     The package id of the app to install
      --version=VALUE        The package version to install; the default is to
                               install the latest version
      ... snip
```

I've added a check for `_all` and `_version` 👍

nblumhardt commented:
Thanks for the great review @liammclennan.

I've updated the help command output to now include examples for all commands, hopefully this improves on the first point, though we could definitely do more here 👍 

```
seqcli app install [<args>]

Install an app package

Example:
  seqcli app install --package-id 'Seq.App.JsonArchive'

Arguments:
      --package-id=VALUE     The package id of the app to install
      --version=VALUE        The package version to install; the default is to
                               install the latest version
      --feed-id=VALUE        The id of the NuGet feed to install the package
                               from; may be omitted if only one feed is
                               configured
...
```

I've added a check to ensure that `--all` and `--version` are exclusive.
